### PR TITLE
EVEREST-278: improve connection errors message

### PR DESCRIPTION
**Pull request by michal-kralik**
EVEREST-278
EVEREST-258

- Check connection to Everest early
- Improve error message during k8s cluster unregistration

### CODE REVIEW

File: pkg/install/operators.go [hunk 1]
```diff
@@ -196,13 +197,34 @@ func (o *Operators) Run(ctx context.Context) error {
...
 	}
 
+	if err := o.checkEverestConnection(ctx); err != nil {
+		var u *url.Error
+		if errors.As(err, &u) {
+			o.l.Debug(err)
+
+			l := o.l.WithOptions(zap.AddStacktrace(zap.DPanicLevel))
+			l.Error("Could not connect to Everest. " +
+				"Make sure Everest is running and is accessible from this computer/server.",
```
Comment (recharte) [inline pkg/install/operators.go:207]: Generic term.
```suggestion
				"Make sure Everest is running and is accessible from this machine.",
```

File: pkg/install/operators.go [hunk 2]
```diff
@@ -196,13 +197,34 @@ func (o *Operators) Run(ctx context.Context) error {
...
 	if err := o.resolveMonitoringInstanceName(ctx); err != nil {
 		return err
 	}
 
 	return o.performProvisioning(ctx)
 }
 
+func (o *Operators) checkEverestConnection(ctx context.Context) error {
+	o.l.Info("Checking connection to Everest")
+	_, err := o.everestClient.ListMonitoringInstances(ctx)
```
Comment (recharte) [inline pkg/install/operators.go:224]: Just food for thought, should we create an endpoint for this?

File: pkg/install/operators.go [hunk 3]
```diff
@@ -388,8 +410,17 @@ func (o *Operators) runMonitoringConfigWizard(ctx context.Context) error {
...
+		var u *url.Error
+		if errors.As(err, &u) {
+			o.l.Debug(err)
+		} else {
+			o.l.Error(err)
+		}
+
+		l := o.l.WithOptions(zap.AddStacktrace(zap.DPanicLevel))
+		l.Error("Could not get a list of monitoring instances from Everest. " +
 			"Make sure Everest is running and is accessible from this computer/server.")
```
Comment (recharte) [inline pkg/install/operators.go:422]: Generic term.
```suggestion
			"Make sure Everest is running and is accessible from this machine.")
```

**Review by recharte — APPROVED** [review]

File: pkg/install/operators.go (hunk 2 above)
Comment (michal-kralik) [reply pkg/install/operators.go:224]: Yes, I think some kind of `/ready` would make sense in the future.

**Review by oksana-grishchenko — APPROVED** [review]

**Pull request merged**